main.go

```diff
@@ -163,6 +165,15 @@ func newClient(cfg *externaldns.Config) (*kubernetes.Clientset, error) {
 		return nil, err
 	}
 
+	config.WrapTransport = func(rt http.RoundTripper) http.RoundTripper {
```

Comment:
Thought about it, too, but then decided to have it always-on. (kube-apiserver etc. also collect metrics by default)